contracts/strategy/BnbxYieldConverterStrategy.sol

```diff
@@ -0,0 +1,245 @@
+    function claimNextBatch() external returns (bool foundClaimableReq) {
+        uint256 claimableIdx;
+        bool foundClaimableReq = false;
+
+        IStakeManager.WithdrawalRequest[] memory requests = _stakeManager
+            .getUserWithdrawalRequests(address(this));
+
+        for (uint256 idx = 0; idx < requests.length; idx++) {
+            (bool isClaimable, uint256 amount) = _stakeManager
+                .getUserRequestStatus(address(this), idx);
+            if (isClaimable) {
+                foundClaimableReq = true;
+                claimableIdx = idx;
+                _bnbToDistribute += amount;
+                break;
+            }
+        }
+
+        if (!foundClaimableReq) return foundClaimableReq;
+
+        _stakeManager.claimWithdraw(claimableIdx);
+        return foundClaimableReq;
+    }
+
+    /// @dev distribute claimed funds to users in FIFO order of withdraw requests
+    /// @param maxNumRequests : parameter to control max number of requests to refund
+    /// @return reqCount : actual number requests refunded
+    function distributeFund(uint256 maxNumRequests)
+        external
+        returns (uint256 reqCount)
+    {
+        reqCount = 0;
+
+        while (
+            _firstWithdrawIdx < _nextWithdrawIdx &&
+            _withdrawRequests[_firstWithdrawIdx].amount <= _bnbToDistribute
+        ) {
+            if (reqCount == maxNumRequests) break;
+            address recipient = _withdrawRequests[_firstWithdrawIdx].recipient;
+            uint256 amount = _withdrawRequests[_firstWithdrawIdx].amount;
+
+            delete _withdrawRequests[_firstWithdrawIdx];
+            _firstWithdrawIdx++;
+
+            _bnbToDistribute -= amount;
+            AddressUpgradeable.sendValue(payable(recipient), amount);
+        }
+    }
+
+    /// @dev claims yeild from stader in BNBx and transfers to feeRecipient
+    function harvest() external onlyStrategist {
+        _harvestTo(rewards);
+    }
+
+    /// @dev internal function to claim yeild from stader in BNBx and transfer them to desired address
+    function _harvestTo(address to) private returns (uint256 yield) {
+        yield = _calculateYield();
+        
+        require(yield > 0, "no yeild to harvest");
+
+        _bnbxHoldingBalance -= yield;
+        _bnbxToken.safeTransfer(to, yield);
+    }
+
+    function _calculateYield() private view returns (uint256 yield) {
+        uint256 bnbxEqAmount = _stakeManager.convertBnbToBnbX(
+            _bnbDepositBalance
+        );
+
+        yield = _bnbxHoldingBalance - bnbxEqAmount;
+    }
+
+    // returns the total amount of tokens in the destination contract
+    function balanceOfPool() public view override returns (uint256) {
+        return _bnbDepositBalance;
+    }
+
+    receive() external payable {}
+
+    function canDeposit(uint256 amount) public pure returns (bool) {
+        return (amount > 0);
```

Comment:
why do you need this check?

Reply:
not required in our case.
It was present in interface, so have to add it.
was also present in the [doc](https://docs.google.com/document/d/1PZT7Dy5CjCdpA7CJcc4JFblberAogllMdyRslXf5kD8/edit) to add it.